src/pydrex/vtk_helpers.py

```diff
@@ -0,0 +1,57 @@
+"""PyDRex: VTK wrappers and helper functions."""
+import pathlib
+
+import numpy as np
+from vtk import vtkXMLUnstructuredGridReader, vtkXMLPUnstructuredGridReader
+
+
+def get_output(filename):
+    """Get a reference to an unstructured (XML) VTK grid stored in `filename`.
+
+    Only supports modern vtk formats, i.e. .vtu and .pvtu files.
+
+    """
+    input_path = pathlib.Path(filename)
+    if input_path.suffix == ".vtu":
+        reader = vtkXMLUnstructuredGridReader()
+    elif input_path.suffix == ".pvtu":
+        reader = vtkXMLPUnstructuredGridReader()
+    else:
+        raise RuntimeError("the supplied input file format is not supported") from None
+
+    reader.SetFileName(filename)
+    reader.Update()
+    return reader.GetOutput()
+
+
+def read_tuple_array(points, fieldname, dimensions):
+    """Read tuples from a vtkPointData object into a numpy array.
+
+    Only reads the specified number of `dimensions` (columns).
+    Throws a LookupError if `fieldname` does not match any data.
+
+    """
+    data = points.GetAbstractArray(fieldname)  # Silently returns None if missing...
+    if data is None:
+        raise LookupError(f"unable to find data for {fieldname}")
+
+    return np.array([data.GetTuple(i) for i in range(data.GetNumberOfTuples())])[
+        :, :dimensions
+    ]
```

Comment:
Does this work for tensor fields?

Reply:
Yes but actually no, I was using it to read e.g. "VelocityGradient" by passing `dimensions = d ** 2` where `d` is 2 or 3. However, this doesn't actually work as I expected (in 2D), because the third column is always full of zeros. I'll need to fix it, do you know if that is specific to fluidity? Or does VTK always use a third dimension full of zeros.

Reply:
OK so I read something about how the third dimension is needed to make things play nice with paraview. That's kinda annoying, since a column of zeros is ambiguous (is it actual data? just a dummy array to appease paraview?) I had this problem when refactoring the original code for my shearbox models because I have a unidirectional velocity field, so something like https://github.com/Patol75/PyDRex/blob/master/src/PyDRex.py#L720 would not work. I will change the code to assume that the third dimension is zero for 2D models.

Reply:
Why would it not work?

Reply:
I mean that I can't use that to detect the dummy columns (not in the coords array but in the others), but anyway I think I've got something cleaner incoming.